client/src/App.svelte

```diff
@@ -5,16 +5,13 @@
   import Home from './view/Home/Home.svelte'
   import { getSocket } from './socket'
   import type { SigninEvArgs } from './type/global'
+  import { getUser } from './utils/userdata'
 
-  let user: { hash?: string; username?: string } = JSON.parse(
-    localStorage.getItem('user') || '{}'
-  )
+  let user = getUser()
   const socket = getSocket(user)
 
   onMount(() => {
-    if (!user.hash) {
-      history.pushState(null, '', '/login')
-    }
+    history.pushState(null, '', !user.hash ? '/login' : '/chat')
```

Comment:
When the user is signed in and refreshes the page, the URL gets set to `/login` even though the view shows the home page. This could lead to confusion, so it's now set to '/chat'.